### PR TITLE
EVE-817 - Setting up EVE pipeline

### PR DESCRIPTION
**Pull request by tcarmet**

### CODE REVIEW

File: eve/main.yml [hunk 1]
```diff
@@ -0,0 +1,48 @@
+---
+version: 0.2
+
+branches:
+  default:
+    stage: pre-merge
+
+models:
```
Comment (sylvain-killian) [inline eve/main.yml]: you don't need models since it is used only once, it would be clearer with a flat yaml

File: eve/main.yml [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    worker: &master-worker
+      type: docker
+      path: eve/workers/master
+      volumes:
+        - '/home/eve/workspace'
+    steps:
+      - Git: *clone
+      - ShellCommand:
+          name: install
+          command: npm install
```
Comment (sylvain-killian) [inline eve/main.yml]: external deps must be included in the docker image to reduce depedency to outside world during the build

File: eve/main.yml [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+      - ShellCommand:
+          name: run test
+          command: npm run --silent test
+      - ShellCommand:
+          name: run ft_test
+          command: npm run ft_test
+      - ShellCommand:
+          name: run executables tests
+          command: |
+            cd lib/executables/pensieveCreds &&
```
Comment (sylvain-killian) [inline eve/main.yml]: you could use workdir ShellCommand option

File: eve/main.yml [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+            ln -s ../../../node_modules . &&
+            npm install && npm test
```
Comment (sylvain-killian) [inline eve/main.yml]: putting the package file in the worker dir, then link it at root, may give better code?

File: eve/workers/master/Dockerfile [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+# Misc
+#
+
+# RUN echo '127.0.0.1    bucketwebsitetester.s3-website-us-east-1.amazonaws.com' >> /etc/hosts
+
+#
+# Run buildbot-worker on startup
+#
+
+ARG BUILDBOT_VERSION=0.9.7
```
Comment (sylvain-killian) [inline eve/workers/master/Dockerfile]: 0.9.12 currently

Comment (tcarmet) [reply eve/workers/master/Dockerfile]: Done

File: eve/main.yml (hunk 3 above)
Comment (tcarmet) [reply eve/main.yml]: Now using workdir

File: eve/main.yml (hunk 1 above)
Comment (tcarmet) [reply eve/main.yml]: I've removed the model

File: eve/main.yml (hunk 4 above)
Comment (tcarmet) [reply eve/main.yml]: In this case, it's another `package.json` file that is called. I've linked the main one which is located on the root directory to the worker. Not sure if will do the same with this one.

File: eve/main.yml (hunk 2 above)
Comment (tcarmet) [reply eve/main.yml]: There's no more npm install, all set with the docker image

File: eve/main.yml (hunk 4 above)
Comment (tcarmet) [reply eve/main.yml]: and btw, this step takes exactly 4 seconds.

**Pull request closed without merging**